hr_announcement/models/res_users.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    @api.model
+    def announcement_user_count(self):
+        # res = super().announcement_user_count()
+        employee_announcements = self.env["announcement"].search_read(
+            [
+                ("announcement_type", "=", "employee"),
+                ("in_date", "=", True),
+                ("id", "not in", self.env.user.read_announcement_ids.ids),
+            ],
+            ["employee_ids"],
+        )
+        announcements = self.env["announcement"].browse(
+            [
+                announcement["id"]
+                for announcement in employee_announcements
+                if announcement["employee_ids"]
+                and any(
+                    id in self.env["hr.employee"].search([("user_id", "!=", False)]).ids
+                    for id in announcement["employee_ids"]
+                )
+            ]
+        )
+        department_announcements = self.env["announcement"].search_read(
+            [
+                ("announcement_type", "=", "department"),
+                ("in_date", "=", True),
+                ("id", "not in", self.env.user.read_announcement_ids.ids),
+            ],
+            ["department_ids"],
+        )
+        departments_with_members = self.env["hr.department"].search(
+            [("member_ids.user_id", "!=", False)]
+        )
+        announcements += self.env["announcement"].browse(
+            [
+                announcement["id"]
+                for announcement in department_announcements
+                if any(
+                    department_id in departments_with_members.ids
+                    for department_id in announcement["department_ids"]
+                )
+            ]
+        )
+
+        position_announcements = self.env["announcement"].search_read(
+            [
+                ("announcement_type", "=", "job_position"),
+                ("in_date", "=", True),
+                ("id", "not in", self.env.user.read_announcement_ids.ids),
+            ],
+            ["position_ids"],
+        )
+        jobs_with_employees = self.env["hr.job"].search(
+            [("employee_ids.user_id", "!=", False)]
+        )
+        announcements += self.env["announcement"].browse(
+            [
+                announcement["id"]
+                for announcement in position_announcements
+                if any(
+                    position_id in jobs_with_employees.ids
+                    for position_id in announcement["position_ids"]
+                )
+            ]
+        )
+
+        return super().announcement_user_count() + [
```

Comment:
This kinda ugly. It's preferable to call to res as you intended and then return res += your_new_results